players/g1_player.py

```diff
@@ -29,12 +31,80 @@ def generate_choices(top_layer: np.ndarray, curr_level: np.ndarray) -> List[Choi
     return res
 
 
+def choose_next_player(now_turn: int, possible_players: List[int], served_situation: List[Dict[int, int]],
+                       top_layer: np.ndarray,
+                       curr_level: np.ndarray, rng: np.random.Generator, group_id: int) -> int:
+    if len(possible_players) == 1:
+        return possible_players[0]
+
+    # if now_turn = 0, it means for the first turn, we just randomly choose a person to pass to
+    if now_turn == 0 and len(possible_players) != 0:
+        return rng.choice(possible_players)
+
+    # if there are no candidates, which means our team is the last one to serve
+    if len(possible_players) == 0:
+        possible_players = [i for i in range(10) if i != group_id]
+
+    choices = generate_choices(top_layer, curr_level)
+    # if all ice cream has been taken
+    if len(choices) == 0:
+        return rng.choice(list(possible_players))
```

Comment:
impossible situation
delete or replace by `assert`

Reply:
actually, I test it, and it truly exists